File: abc130/abc130_b.cs
// https://atcoder.jp/contests/abc130/submissions/8296528

using System;
using System.Linq;

namespace CSharp
{
    class Program
    {
        static void Main()
        {
            var input = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
            var N = input[0];
            var X = input[1];

            var L = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();

            var sum = 0;
            for (int i = 0; i < N; i++)
            {
                if ((sum += L[i]) > X)
                {
                    Console.WriteLine(i + 1);
                    return;
                }
            }
            Console.WriteLine(N + 1);
        }
    }
}

File: abc/abc130/abc130_b.fs
// https://atcoder.jp/contests/abc130/submissions/11959532

let [| N; X |] = stdin.ReadLine().Split() |> Array.map int
let L = stdin.ReadLine().Split() |> Array.map int

let rec solver s n =
    if s > X then n
    elif n = N then N + 1
    else solver (s + L.[n]) (n + 1)

solver 0 0 |> stdout.WriteLine
